src/passes/Inlining.cpp

```diff
@@ -146,7 +146,20 @@ struct Planner : public WalkerPass<PostWalker<Planner>> {
     // plan to inline if we know this is valid to inline, and if the call is
     // actually performed - if it is dead code, it's pointless to inline.
     // we also cannot inline ourselves.
-    if (state->worthInlining.count(curr->target) && curr->type != unreachable &&
+    bool isUnreachable;
+    if (curr->isReturn) {
+      // Tail calls are only actually unreachable if an argument is
+      isUnreachable = false;
+      for (auto* op : curr->operands) {
+        if (op->type == unreachable) {
+          isUnreachable = true;
+          break;
+        }
+      }
```

Comment:
I feel this logic should be in `ir/` as a utility function. Perhaps in a file for utilities for `ExpressionList`s (which doesn't exist yet). Or maybe is there some functional programming-style way to write this in modern C++?

Reply:
This would be a good job for `std::any_of`, which I just learned exists 🎉